File: Data Structures & Algorithms/is-anagram/submission-4.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        seen = {}
        n = len(s)
        if n != len(t):
            return False
        
        for i in range(n):
            if s[i] not in seen:
                seen[s[i]] = 1
            else: 
                seen[s[i]] += 1
        
        for i in range(n):
            if t[i] not in seen or seen[t[i]] == 0:
                return False
            seen[t[i]] -= 1
        
        return True

File: Data Structures & Algorithms/contains-duplicate-ii/submission-4.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_set<int> seen;
        int n = nums.size();
        for(int i=0;i<n;i++){
            int j = i+1;
            while(j<n && abs(j-i) <= k){
                if(nums[j] == nums[i]){
                    return true;
                }
                j++;
            }

            seen.erase(nums[j-i]);
        }
        return false;
    }
};
